tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py

```diff
@@ -0,0 +1,190 @@
+        pretrained = self.config.get("pretrained", "laion400m_e32")
+
+        self.load_models(network_info, launcher, True)
+
+        self.text_encoder = launcher.ie_core.compile_model(self.text_encoder_model, launcher.device)
+        self.image_encoder = launcher.ie_core.compile_model(self.image_encoder_model, launcher.device)
+
+        unet_shapes = [inp.get_partial_shape() for inp in self.text_encoder_model.inputs]
+        if unet_shapes[0][0].is_dynamic:
+            self.templates_file = self.config.get("templates", "zeroshot_classification_templates.json")
+        else:
+            self.templates_file = None
+
+        self.classnames_file = self.config.get("classnames", "classnames.json")
+
+        self.torch_model, _, self.transform = open_clip.create_model_and_transforms(orig_model_name,
+                                                                                    pretrained=pretrained,
+                                                                                    cache_dir=None)
+        self.tokenizer = open_clip.get_tokenizer(orig_model_name)
+
+    def predict(self, identifiers, input_data, zeroshot_weights):
+        preds = []
+        for idx, image_data in zip(identifiers, input_data):
+            image = self.transform(PIL.Image.fromarray(image_data))
+            image_features = self.encode_image(image.unsqueeze(0))
+            image_features = F.normalize(image_features, dim=-1)
+            logits = 100. * image_features @ zeroshot_weights
+            preds.append(ClassificationPrediction(idx, logits.squeeze(0)))
+        return None, preds
+
+    def get_network(self):
+        models = self.pipe.get_models()
+        model_list = []
+        for model_part_name, model in models.items():
+            model_list.append({"name": model_part_name, "model": model})
+        return model_list
+
+    def encode_image(self, image, normalize: bool = False):
+        '''
+        Code source https://github.com/AlexKoff88/CLIP_benchmark/blob/openvino/clip_benchmark/models/openvino_clip.py
+        '''
+        features = self.image_encoder(image)
+        features = torch.from_numpy(features[self.image_encoder.output()])
+        return F.normalize(features, dim=-1) if normalize else features
+
+    def encode_text(self, text, normalize: bool = False):
+        # Code source https://github.com/AlexKoff88/CLIP_benchmark/blob/openvino/clip_benchmark/models/openvino_clip.py
+        x = self.torch_model.token_embedding(text)
+        x = x + self.torch_model.positional_embedding
+        x = x.permute(1, 0, 2)
+        x = torch.asarray(x).contiguous()
+        x = self.text_encoder((x, self.torch_model.attn_mask))
+        x = torch.from_numpy(x[self.text_encoder.output()])
+        x = x.permute(1, 0, 2)
+        x = self.torch_model.ln_final(x)
+        x = x[np.arange(x.shape[0]), text.argmax(dim=-1)] @ self.torch_model.text_projection
+        return F.normalize(x, dim=-1) if normalize else x
```

Comment:
I do not think that it is good to introduce torch dependency for operations that can be done using numpy

Reply:
Rewritten to not use torch